### PR TITLE
WHAN-33 ResourceMedia supports request attributes 'widths' and 'sizes'

### PR DESCRIPTION
**Pull request by mrozati**
ResourceMedia supports request attributes 'widths' (array of responsive widths) and 'sizes' (image sizes string, optional)

### CODE REVIEW

File: media/changes.xml [hunk 1]
```diff
@@ -36,6 +36,9 @@
       <action type="remove" dev="sseifert">
         DamVideoMediaMarkupBuilder: Do not generate flash player as fallback.
       </action>
+      <action type="update" dev="mrozati" issue="WHAN-33">
```
Comment (stefanseifert) [inline media/changes.xml]: please rebase with current develop branch and create a new section for release 1.9.0 - 1.8.2 was already released.

File: media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java [hunk 2]
```diff
@@ -70,8 +83,51 @@
...
+   * To be used together with 'imageSizes' property. <br>
+   * Example: "{@literal 2560:false,1920:false,1280:false,640:false,320:false}" <br>
+   * The boolean flag indicates whether the with is required.<br>
+   * Cannot be used together with the picture source parameters.
+   */
+  @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
+  private String widthOptions;
+
+  /**
+   * "Sizes" string for img element.
```
Comment (stefanseifert) [inline media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java]: example value would be helpful.

File: media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java [hunk 3]
```diff
@@ -70,8 +83,51 @@
...
+   * Cannot be used together with the picture source parameters.
+   */
+  @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
+  private String imageSizes;
+
+  /**
+   * List of media formats for the picture source elements.
```
Comment (stefanseifert) [inline media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java]: example value(s) would be helpful.

File: media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java [hunk 4]
```diff
@@ -70,8 +83,51 @@
...
+   * You have to define the same number of array items in all pictureSource* properties.
+   * Cannot be used together with image sizes.
+   */
+  @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
+  private Object[] pictureSourceMediaFormat;
+
+  /**
+   * List of media expressions for the picture source elements.
```
Comment (stefanseifert) [inline media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java]: example value(s) would be helpful.

File: media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java [hunk 5]
```diff
@@ -70,8 +83,51 @@
...
+   * You have to define the same number of array items in all pictureSource* properties.
+   * Cannot be used together with image sizes.
+   */
+  @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
+  private Object[] pictureSourceMedia;
+
+  /**
+   * List of widths for the picture source elements.
```
Comment (stefanseifert) [inline media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java]: example value(s) would be helpful.

File: media/src/main/java/io/wcm/handler/media/util/ImageUtils.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) pro!vision GmbH. All rights reserved. */
```
Comment (stefanseifert) [inline media/src/main/java/io/wcm/handler/media/util/ImageUtils.java]: please apply the usual wcm.io apache license file header (with 2020 as date)

File: media/src/main/java/io/wcm/handler/media/util/ImageUtils.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+package io.wcm.handler.media.util;
```
Comment (stefanseifert) [inline media/src/main/java/io/wcm/handler/media/util/ImageUtils.java]: this creates a new package that is exported by OSGi - but i suppose the util class is not intended for public usage (we can always make it public later).
recommendation: move the class to {{io.wcm.handler.media.ui}} and make it non-public there.

Comment (mrozati) [reply media/src/main/java/io/wcm/handler/media/util/ImageUtils.java]: good point. I moved it

File: media/changes.xml (hunk 1 above)
Comment (mrozati) [reply media/changes.xml]: done

File: media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java (hunk 2 above)
Comment (mrozati) [reply media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java]: done

File: media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java (hunk 3 above)
Comment (mrozati) [reply media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java]: done

File: media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java (hunk 5 above)
Comment (mrozati) [reply media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java]: done

File: media/src/main/java/io/wcm/handler/media/util/ImageUtils.java (hunk 6 above)
Comment (mrozati) [reply media/src/main/java/io/wcm/handler/media/util/ImageUtils.java]: fixed

**Pull request merged**